Code/Framework/AzCore/Tests/Memory/AllocatorBenchmarks.cpp

```diff
@@ -0,0 +1,590 @@
+        {
+            AZ::AllocatorInstance<TAllocator>::Get().DeAllocate(ptr, byteSize);
+        }
+
+        static void* ReAllocate(void* ptr, size_t newSize, size_t newAlignment)
+        {
+            return AZ::AllocatorInstance<TAllocator>::Get().ReAllocate(ptr, newSize, newAlignment);
+        }
+
+        static size_t Resize(void* ptr, size_t newSize)
+        {
+            return AZ::AllocatorInstance<TAllocator>::Get().Resize(ptr, newSize);
+        }
+
+        static void GarbageCollect()
+        {
+            AZ::AllocatorInstance<TAllocator>::Get().GarbageCollect();
+        }
+
+        static size_t NumAllocatedBytes()
+        {
+            return AZ::AllocatorInstance<TAllocator>::Get().NumAllocatedBytes() +
+                AZ::AllocatorInstance<TAllocator>::Get().GetUnAllocatedMemory();
+        }
+
+        static size_t GetSize(void* ptr)
+        {
+            return AZ::AllocatorInstance<TAllocator>::Get().AllocationSize(ptr);
+        }
+    };
+
+    /// <summary>
+    /// Basic allocator used as a baseline. This allocator is the most basic allocation possible with the OS (AZ_OS_MALLOC).
+    /// MallocSchema cannot be used here because it has extra logic that we don't want to use as a baseline.
+    /// </summary>
+    class RawMallocAllocator {};
+
+    template<>
+    class TestAllocatorWrapper<RawMallocAllocator>
+    {
+    public:
+        TestAllocatorWrapper()
+        {
+            s_numAllocatedBytes = 0;
+        }
+
+        static void SetUp()
+        {
+            s_numAllocatedBytes = 0;
+        }
+
+        static void TearDown()
+        {
+        }
+
+        static void* Allocate(size_t byteSize, size_t)
+        {
+            s_numAllocatedBytes += byteSize;
+            // Don't pass an alignment since we wont be able to get the memory size without also passing the alignment
+            return AZ_OS_MALLOC(byteSize, 1);
+        }
+
+        static void DeAllocate(void* ptr, size_t = 0)
+        {
+            s_numAllocatedBytes -= Platform::GetMemorySize(ptr);
+            AZ_OS_FREE(ptr);
+        }
+
+        static void* ReAllocate(void* ptr, size_t newSize, size_t)
+        {
+            s_numAllocatedBytes -= Platform::GetMemorySize(ptr);
+            AZ_OS_FREE(ptr);
+
+            s_numAllocatedBytes += newSize;
+            return AZ_OS_MALLOC(newSize, 1);
+        }
+
+        static size_t Resize(void* ptr, size_t newSize)
+        {
+            AZ_UNUSED(ptr);
+            AZ_UNUSED(newSize);
+
+            return 0;
+        }
+
+        static void GarbageCollect() {}
+
+        static size_t NumAllocatedBytes()
+        {
+            return s_numAllocatedBytes;
+        }
+
+        static size_t GetSize(void* ptr)
+        {
+            return Platform::GetMemorySize(ptr);
+        }
+
+    private:
+        static size_t s_numAllocatedBytes;
+    };
+
+    size_t TestAllocatorWrapper<RawMallocAllocator>::s_numAllocatedBytes = 0;
```

Comment:
You could use the `inline` keyword to initialize this to 0.
```suggestion
        inline static size_t s_numAllocatedBytes = 0;
    };

```